iothub/service/src/Amqp/AmqpConnectionHandler.cs

```diff
@@ -95,26 +95,47 @@ internal async Task OpenAsync(CancellationToken cancellationToken)
 
                 AmqpSettings amqpSettings = CreateAmqpSettings();
 
+                AmqpTransportInitiator amqpTransportInitiator;
                 if (_useWebSocketOnly)
                 {
-                    // Try only AMQP transport over WebSocket
-                    _transport = await CreateClientWebSocketTransportAsync(cancellationToken).ConfigureAwait(false);
+                    var websocketUri = new Uri($"{AmqpsConstants.Scheme}{_credential.HostName}:{AmqpsConstants.SecurePort}{AmqpsConstants.UriSuffix}");
+                    var websocketTransportSettings = new WebSocketTransportSettings
+                    {
+                        Uri = websocketUri,
+                        Proxy = _options.Proxy,
+                        SubProtocol = AmqpsConstants.Amqpwsb10,
+                    };
+
+                    amqpTransportInitiator = new AmqpTransportInitiator(amqpSettings, websocketTransportSettings);
```

Comment:
Doesn't the websocket connection require/accept the certificate information and the remote certificate validation callback? I was expecting the tls settings to be common for both.

Reply:
This appears to be a gap in feature parity between TCP and Websocket in the AMQP library. I'll investigate this a bit more

Reply:
So this is the problem that we've already discussed as a team. Currently, the AMQP library doesn't allow us to pass in the client web socket instance. Since the AMQP library targets .netstandard2.0 which doesn't support setting a remote certificate validation callback, there is no way for us to pass in just the callback for the client websocket to use.

To get around this, we can ask the AMQP library to take in the client web socket instance. That way, users can construct on themselves and set the remote certificate validation callback on that instance.

https://github.com/Azure/azure-amqp/pull/226

Reply:
We currently don't support setting this option on AMQP web sockets, though, so this PR isn't breaking anything in that regard